sdk/nodejs/automation/localWorkspace.ts

```diff
@@ -1062,33 +1089,31 @@ function loadProjectSettings(workDir: string) {
  *
  * @param minVersion The minimum acceptable version of the Pulumi CLI.
  * @param currentVersion The currently known version. `null` indicates that the current version is unknown.
- * @paramoptOut If the user has opted out of the version check.
+ * @param optOut If the user has opted out of the version check.
  */
-export function parseAndValidatePulumiVersion(
+export function validatePulumiVersion(
```

Comment:
Workspace is always instantiating a `Pulumi` wrapper for the command and the constructor for `Pulumi` does a version check already. I think it would make sense to move these checks into that class.

This check should also respect the `PULUMI_AUTOMATION_API_SKIP_VERSION_CHECK` env variable.